lineapy/system_tracing/_op_stack.py

```diff
@@ -16,18 +24,38 @@ class Frame(Structure):
     object, so we can access the top of the stack.
     """
 
-    _fields_: Tuple[Tuple[str, object], ...] = (
-        ("ob_refcnt", c_ssize_t),
-        ("ob_type", c_void_p),
-        ("ob_size", c_ssize_t),
-        ("f_back", c_void_p),
-        ("f_code", c_void_p),
-        ("f_builtins", POINTER(py_object)),
-        ("f_globals", POINTER(py_object)),
-        ("f_locals", POINTER(py_object)),
-        ("f_valuestack", POINTER(py_object)),
-        ("f_stacktop", POINTER(py_object)),
-    )
+    if sys.version_info < (3, 10):
+        _fields_: Tuple[Tuple[str, object], ...] = (
+            ("ob_refcnt", c_ssize_t),
+            ("ob_type", c_void_p),
+            ("ob_size", c_ssize_t),
+            ("f_back", c_void_p),
+            ("f_code", c_void_p),
+            ("f_builtins", POINTER(py_object)),
+            ("f_globals", POINTER(py_object)),
+            ("f_locals", POINTER(py_object)),
+            ("f_valuestack", POINTER(py_object)),
+            ("f_stacktop", POINTER(py_object)),
+            ("f_trace", POINTER(py_object)),
+        )
+    elif (3, 11) > sys.version_info >= (3, 10):
+        _fields_: Tuple[Tuple[str, object], ...] = (
+            ("ob_refcnt", c_ssize_t),
+            ("ob_type", c_void_p),
+            ("ob_size", c_ssize_t),
+            ("f_back", c_void_p),
+            ("f_code", c_void_p),
+            ("f_builtins", POINTER(py_object)),
+            ("f_globals", POINTER(py_object)),
+            ("f_locals", POINTER(py_object)),
+            ("f_valuestack", POINTER(py_object)),
+            ("f_trace", POINTER(py_object)),
+            ("f_stackdepth", c_int),
+        )
+    else:
+        raise NotImplementedError(
+            "LineaPy currently does not support Python 3.11 and above"
+        )
```

Comment:
Might be nice to get a sense of what we don't support in 3.11 instead of eagerly crashing. Also if we want to eagerly crash, we should do so at setup time, instead of sometime into the session (bad UX).

Reply:
Updated